src/main/java/org/gaul/s3proxy/Main.java

```diff
@@ -271,6 +272,19 @@ private static BlobStore parseMiddlewareProperties(BlobStore blobStore,
                 properties);
         }
 
+        var storageClass = properties.getProperty(
+                S3ProxyConstants.PROPERTY_STORAGE_CLASS_BLOBSTORE);
+        if (!Strings.isNullOrEmpty(storageClass)) {
+            System.err.println("Using storage class override backend");
+            var storageClassBlobStore =
+                    StorageClassBlobStore.newStorageClassBlobStore(
+                            blobStore, storageClass);
+            blobStore = storageClassBlobStore;
+            System.err.println("Requested tier: tier");
+            System.err.println("Overriden to: " +
+                    StorageClass.fromTier(storageClassBlobStore.getTier()));
```

Comment:
This only works for S3-S3 mappings and is confusing for other storage backends.

Reply:
I changed the wiki accordingly.